lib/parallel_tests/grouper.rb

```diff
@@ -28,7 +28,39 @@ def in_even_groups_by_size(items, num_groups, options = {})
           raise 'Number of isolated processes must be less than total the number of processes'
         end
 
-        if isolate_count >= 1
+        if options[:specify_groups]
+          specify_test_process_groups = options[:specify_groups].split('|')
+          all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+          if specify_test_process_groups.count > num_groups
+            raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+          end
+
+          specified_items_found, items = items.partition do |item, _size|
+            all_specified_tests.any? { |specified_spec| item == specified_spec }
+          end
+
+          specified_specs_not_found = all_specified_tests - specified_items_found.map(&:first)
```

Comment:
```suggestion
          specified_specs_not_found = all_specified_tests - specified_items_found.keys
```

Reply:
I don't think we can do `.keys` here because its an array of arrays like this
```ruby
[["spec/test1.rb", 1], ["spec/test2.rb", 2], ["spec/test3.rb", 3]]
```

Reply:
it will be a hash with the suggested change above

Reply:
true...dang I think I need to update some of the unit tests...I dont think their data structures match what they would be at this stage...

Reply:
I take it back, I think items at this point should be an array of arrays like this
```ruby
[["spec/parallel_tests/cli_spec.rb", 15103],
 ["spec/parallel_tests/cucumber/failure_logger_spec.rb", 1676],
 ["spec/parallel_tests/cucumber/runner_spec.rb", 2508],
 ["spec/parallel_tests/cucumber/scenarios_spec.rb", 7811],
 ["spec/parallel_tests/grouper_spec.rb", 5028],
 ["spec/parallel_tests/pids_spec.rb", 706],
 ["spec/parallel_tests/rspec/failures_logger_spec.rb", 409],
 ["spec/parallel_tests/rspec/logger_base_spec.rb", 815],
 ["spec/parallel_tests/rspec/runner_spec.rb", 6801],
 ["spec/parallel_tests/rspec/runtime_logger_spec.rb", 3358],
 ["spec/parallel_tests/rspec/summary_logger_spec.rb", 429],
 ["spec/parallel_tests/spinach/runner_spec.rb", 403],
 ["spec/parallel_tests/tasks_spec.rb", 7345],
 ["spec/parallel_tests/test/runner_spec.rb", 21006],
 ["spec/parallel_tests/test/runtime_logger_spec.rb", 1245],
 ["spec/rails_spec.rb", 1464]]
```